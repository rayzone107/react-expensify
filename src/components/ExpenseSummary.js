import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ count, total }) => {
    const expenseWord = count === 1 ? "expense" : "expenses";
    const expenseTotal = numeral(total / 100).format('$0,0.00');
    return (
        <div>
            {
                count > 0 && (
                    <p>Viewing {count} {expenseWord} totalling {expenseTotal}</p>
                )
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        count: visibleExpenses.length,
        total: getExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);
