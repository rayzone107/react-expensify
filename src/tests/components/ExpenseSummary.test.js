import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from "../../components/ExpenseSummary";
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary count={1} total={250} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary count={3} total={49520} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with no expenses', () => {
    const wrapper = shallow(<ExpenseSummary count={0} total={0} />);
    expect(wrapper).toMatchSnapshot();
});
