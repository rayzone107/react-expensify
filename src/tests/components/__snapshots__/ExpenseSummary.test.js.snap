// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.50
  </p>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $495.20
  </p>
</div>
`;

exports[`should render ExpenseSummary with no expenses 1`] = `<div />`;
